### ISSUE TITLE
`no-redundant-story-name` case insensitivity

### ISSUE THREAD

**Issue by benedfit** [storybookjs/eslint-plugin-storybook#94]
**Is your feature request related to a problem? Please describe.**
Warn/error when `storyName` is a case-insensitive variant of the name generated by the export name:

e.g. `WithText.storyName = 'With Text';` will currently warn/error but `WithText.storyName = 'with text';` will not

**Describe the solution you'd like**
The ability to configure the `no-redundant-story-name` rule to use/ignore case when determining in the name is redundant

**Describe alternatives you've considered**
Would also be open to thing being the default behaviour if that's easier.

**Comment by yannbf** [comment]
Hey @benedfit thanks for opening this issue! I am not sure how valid this request is, given that there is a difference between the scenarios which you described:
1. `WithText.storyName = 'With Text';`
2. `WithText.storyName = 'with text';`

They will yield different results in the sidebar. People might actually want to specify names to be lower case, whilst the story name is upper case.

**Comment by benedfit** [comment]
@yannbf I agree that there would be scenarios where folks want to specify lower case, so that's why it would be great for it to be an optional settings, and for it to work as it currently does by default
